mmv1/third_party/terraform/services/apphub/data_source_apphub_discovered_workload_test.go

```diff
@@ -0,0 +1,112 @@
+package apphub_test
+
+import (
+    "testing"
+
+    "github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+    "github.com/hashicorp/terraform-provider-google/google/acctest"
+)
+
+func TestAccDataSourceApphubDiscoveredWorkload_basic(t *testing.T) {
+    t.Parallel()
+
+    context := map[string]interface{}{
+    	"random_suffix": acctest.RandString(t, 10),
+    }
+
+    acctest.VcrTest(t, resource.TestCase{
+        PreCheck:                 func() { acctest.AccTestPreCheck(t) },
+        ProtoV5ProviderFactories: acctest.ProtoV5ProviderFactories(t),
+        Steps: []resource.TestStep{
+            {
+                Config: testDataSourceApphubDiscoveredWorkload_basic(context),
+            },
+        },
+    })
+}
+
+func testDataSourceApphubDiscoveredWorkload_basic(context map[string]interface{}) string {
+    return acctest.Nprintf(`
+data "google_apphub_discovered_workload" "catalog-workload" {
+  provider = google
+  location = "us-central1"
+  workload_uri = "${replace(google_compute_region_instance_group_manager.mig.instance_group, "https://www.googleapis.com/compute/v1", "//compute.googleapis.com")}"
+  depends_on = [google_compute_region_instance_group_manager.mig]
+}
+
+# VPC network
+resource "google_compute_network" "ilb_network" {
+  name                    = "l7-ilb-network-%{random_suffix}"
+  project                 = "tf-project-5-416803"
```

Comment:
Why is this hard-coded?
Consider creating a google_project, performing a service project attachment using https://github.com/GoogleCloudPlatform/magic-modules/blob/ab2122202ab62d620cbbbc8928078b0782146994/mmv1/products/apphub/ServiceProjectAttachment.yaml; and create all resources on the service project.

Reply:
Done.